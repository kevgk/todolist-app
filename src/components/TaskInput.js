import React, { useRef, useContext, useEffect } from 'react';
import { TodoContext } from '../store/todo';
import localStorageSaveJSON from '../utils/localStorageSaveJSON';

export default function TaskInput() {
  const taskInput = useRef();
  const { state, dispatch } = useContext(TodoContext);

  const isFirstRun = useRef(true);

  const handleSubmit = e => {
    e.preventDefault();

    if (taskInput.current.value.length) {
      dispatch({ type: 'ADD_TODO', payload: { name: taskInput.current.value }});
      taskInput.current.value = '';
    }
  }

  useEffect(() => {
    if (isFirstRun.current) {
      isFirstRun.current = false;
      return;
    }
    localStorageSaveJSON('tasks', state);
  }, [state]);

  return (
    <form className='taskContainer' onSubmit={e => handleSubmit(e)}>
      <input className='taskInput' autoFocus placeholder="Add a task..." ref={taskInput}/>
    </form>
  );
}