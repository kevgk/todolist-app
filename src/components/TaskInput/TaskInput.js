import React, { useRef, useContext } from 'react';
import { TodoContext } from '../../store/todo';
import styles from './TaskInput.module.css';

export default function TaskInput() {
  const taskInput = useRef();
  const { dispatch } = useContext(TodoContext);

  const handleSubmit = e => {
    e.preventDefault();

    if (taskInput.current.value.length) {
      dispatch({ type: 'ADD_TODO', payload: { name: taskInput.current.value }});
      taskInput.current.value = '';
    }
  }

  return (
    <form className={styles.container} onSubmit={e => handleSubmit(e)}>
      <input className={styles.taskInput} autoFocus placeholder="Add a task..." ref={taskInput}/>
    </form>
  );
}