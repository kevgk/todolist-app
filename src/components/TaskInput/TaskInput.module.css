.container {
  justify-self: flex-end;
  margin-top: auto;
}

.taskInput {
  box-sizing: border-box;
  border-radius: 10px;
  background-color: #7047f5;
  color: #fff;
  border: none;
  padding: 10px;
  width: 100%;
  text-align: center;
  font-size: 15px;
}

.taskInput::placeholder {
  color: rgba(255, 255, 255, 0.5);
}

.taskInput::selection {
  background: #fff;
  color: #7047f5;
}

.taskInput:focus {
  box-shadow: 0 0 10px #7047f5;
}

