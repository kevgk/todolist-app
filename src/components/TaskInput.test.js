import React from 'react';
import { mount } from 'enzyme';

import TaskInput from './TaskInput';

it('renders correctly', () => {
	const wrapper = mount(<TaskInput dispatch={() => {}} />);

	expect(wrapper).toMatchSnapshot();
});
