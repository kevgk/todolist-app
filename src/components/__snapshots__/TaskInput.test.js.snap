// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="taskContainer"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="taskInput"
    placeholder="Add a task..."
  />
</form>
`;
